File: 53_maximum_subarray.py
class Solution:
    def maxSubArray(self, nums) -> int:
        global_max = nums[0]
        curr_max = float('-inf')
        for num in nums:
            curr_max = max(num, curr_max + num)
            global_max = max(global_max, curr_max)
        return global_max

File: leetcode/53_maximum_subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function (nums) {
  let globalMax = nums[0];
  let currMax = Number.MIN_SAFE_INTEGER;
  for (let num of nums) {
    currMax = Math.max(num, currMax + num);
    if (currMax > globalMax) globalMax = currMax;
  }
  return globalMax;
};